markdown/util.py

```diff
@@ -298,39 +301,47 @@ class Registry:
     """
 
     def __init__(self):
-        self._data = {}
+        self._data: dict[str, _T] = {}
         self._priority = []
         self._is_sorted = False
 
-    def __contains__(self, item):
+    def __contains__(self, item: str | _T) -> bool:
         if isinstance(item, str):
             # Check if an item exists by this name.
             return item in self._data.keys()
         # Check if this instance exists.
         return item in self._data.values()
 
-    def __iter__(self):
+    def __iter__(self) -> Iterator[_T]:
         self._sort()
         return iter([self._data[k] for k, p in self._priority])
 
-    def __getitem__(self, key):
+    @overload
+    def __getitem__(self, key: str | int) -> _T:
```

Comment:
Could you please add a ` # pragma: no cover` comment to these overloaded methods?